### ISSUE TITLE
Required validator is broken for `Likert`

### ISSUE THREAD

**Issue by bjosttveit** [Altinn/app-lib-dotnet#513]
### Description of the bug

Likert is a special case, since it has an answer binding that points to a field in a list of objects. Required in this case, means that all of the matching fields must not be empty, instead of the direct binding itself. This is not taken into account by `RunLayoutValidationsForRequiredRecurs` which simply gets the data at the exact path defined in the data-model binding. This works for components inside of repeating groups due to the row-context, but this does not exist for `Likert` "rows".

Likert likely needs its own special case for handling required validations.

Slack: https://altinn.slack.com/archives/C02EJ9HKQA3/p1710493234216779

### Steps To Reproduce

.

**Comment by ivarne** [comment]
Given that the questions are in the data model, this will only ever be a consistency validation, and not a proper validation that the intended questions are answered correctly (as the user have the functionality to add or remove questions).

Does frontend properly handle `"required": true` on a likert component? Does it make sense to just ignore required in backend validation for likert, or do we need a full implementation? There is also an issue with `"filter"` that makes this component pretty complicated to handle.

**Comment by bjosttveit** [comment]
Yeah I guess its true that the user could remove questions since they live in the data model, so maybe we could ignore required validations in Likert for now.

To answer the other questions: `"required": true` is handled properly in the frontend. As for `"filter"`, it can only be used to specify start/stop index where the rest of the questions are ignored (not shown to the user). So it should not be terribly complicated to handle, at least not as bad as the old `"filter"` property for repeating groups.

```json
"filter": [
  {
    "key": "start",
    "value": "0"
  },
  {
    "key": "stop",
    "value": "3"
  }
],
```

**Comment by ivarne** [comment]
Started work on likert component for backend here: https://github.com/Altinn/app-lib-dotnet/compare/ivarne/likert-component?expand=1
